apiserver/facades/agent/meterstatus/meterstatus.go

```diff
@@ -78,72 +73,84 @@ func NewMeterStatusAPI(
+		watcherID := m.resources.Register(w)
+
+		result.Results[i] = params.NotifyWatchResult{
+			NotifyWatcherId: watcherID,
 		}
-		result.Results[i].NotifyWatcherId = watcherId
-		result.Results[i].Error = apiservererrors.ServerError(err)
 	}
-	return result, nil
-}
 
-func (m *MeterStatusAPI) watchOneUnitMeterStatus(tag names.UnitTag) (string, error) {
-	unit, err := m.state.Unit(tag.Id())
-	if err != nil {
-		return "", err
-	}
-	watch := unit.WatchMeterStatus()
-	if _, ok := <-watch.Changes(); ok {
-		return m.resources.Register(watch), nil
-	}
-	return "", watcher.EnsureErr(watch)
+	return result, nil
 }
 
 // GetMeterStatus returns meter status information for each unit.
+// This is a noop as of 3.6.10 because meter status functionality is removed.
 func (m *MeterStatusAPI) GetMeterStatus(args params.Entities) (params.MeterStatusResults, error) {
-	result := params.MeterStatusResults{
+	return params.MeterStatusResults{
 		Results: make([]params.MeterStatusResult, len(args.Entities)),
+	}, nil
+}
+
+// newEmptyNotifyWatcher returns starts and returns a new empty notify watcher,
+// with only the initial event.
+func newEmptyNotifyWatcher() *emptyNotifyWatcher {
+	changes := make(chan struct{})
+
+	w := &emptyNotifyWatcher{
+		changes: changes,
 	}
-	canAccess, err := m.accessUnit()
-	if err != nil {
-		return params.MeterStatusResults{}, apiservererrors.ErrPerm
-	}
-	for i, entity := range args.Entities {
-		unitTag, err := names.ParseUnitTag(entity.Tag)
-		if err != nil {
-			result.Results[i].Error = apiservererrors.ServerError(apiservererrors.ErrPerm)
-			continue
-		}
-		err = apiservererrors.ErrPerm
-		var status state.MeterStatus
-		if canAccess(unitTag) {
-			var unit *state.Unit
-			unit, err = m.state.Unit(unitTag.Id())
-			if err == nil {
-				status, err = MeterStatusWrapper(unit.GetMeterStatus)
-			}
-			result.Results[i].Code = status.Code.String()
-			result.Results[i].Info = status.Info
+	w.tomb.Go(func() error {
+		changes <- struct{}{}
+		defer close(changes)
+		return w.loop()
+	})
+
+	return w
+}
+
+// emptyNotifyWatcher implements watcher.NotifyWatcher.
+type emptyNotifyWatcher struct {
+	changes chan struct{}
+	tomb    tomb.Tomb
+}
+
+// Changes returns the event channel for the empty notify watcher.
+func (w *emptyNotifyWatcher) Changes() <-chan struct{} {
+	return w.changes
+}
+
+// Kill asks the watcher to stop without waiting for it do so.
+func (w *emptyNotifyWatcher) Kill() {
+	w.tomb.Kill(nil)
+}
+
+// Stop asks the watcher to stop and waits.
+func (w *emptyNotifyWatcher) Stop() error {
+	w.tomb.Kill(nil)
+	return w.tomb.Wait()
+}
+
+// Wait waits for the watcher to die and returns any
+// error encountered when it was running.
+func (w *emptyNotifyWatcher) Wait() error {
+	return w.tomb.Wait()
+}
+
+func (w *emptyNotifyWatcher) loop() error {
+	for {
+		select {
+		case <-w.tomb.Dying():
+			return tomb.ErrDying
 		}
-		result.Results[i].Error = apiservererrors.ServerError(err)
 	}
-	return result, nil
 }
```

Comment:
The loop is not required, since it will never loop.
```suggestion
func (w *emptyNotifyWatcher) loop() error {
	<-w.tomb.Dying()
	return tomb.ErrDying
}
```

Reply:
Addressed. Thank you.